### ISSUE TITLE
[Bug]: Release collection: find no available querycoord, check querycoord state after updating config and waiting healthy

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#31628]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.4-20240320-7f201e80-amd64
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    kafka
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.4.0rc53
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus and disable L0 segment
```
fouram-op-10-4140-milvus-standalone-7845cb4b9b-rnvbt     Running     0            1m      10.104.15.72      4am-node20     
fouram-op-10-4140-etcd-0                                 Running     0            2m      10.104.26.123     4am-node32     
fouram-op-10-4140-kafka-0                                Running     2            2m      10.104.29.202     4am-node35     
fouram-op-10-4140-kafka-1                                Running     2            2m      10.104.15.64      4am-node20     
fouram-op-10-4140-kafka-2                                Running     1            2m      10.104.31.221     4am-node34     
fouram-op-10-4140-kafka-zookeeper-0                      Running     0            2m      10.104.31.214     4am-node34     
fouram-op-10-4140-kafka-zookeeper-1                      Running     0            2m      10.104.29.203     4am-node35     
fouram-op-10-4140-kafka-zookeeper-2                      Running     0            2m      10.104.15.65      4am-node20     
fouram-op-10-4140-minio-64d48475cf-w9xtb                 Running     0            2m      10.104.31.216     4am-node34  
```
2. test-before-upgrade:
  create collection -> index -> load -> insert 20m data -> flush -> index -> load -> concurrent upsert + flush + search
3. upgrade milvus config: enable L0 segment -> waiting for running -> release collection failed:
```
[2024-03-26 10:46:15,749 -  INFO - fouram]: [get_pods] pod details of release(fouram-op-10-4140):  (operator_client.py:306)
[2024-03-26 10:46:15,749 -  INFO - fouram]: 
NAME                                                     STATUS      RESTARTS     AGE       IP                NODE           
fouram-op-10-4140-milvus-standalone-54897698d6-cjcsd     Running     0            2m        10.104.16.15      4am-node21     
fouram-op-10-4140-etcd-0                                 Running     0            2h36m     10.104.26.123     4am-node32     
fouram-op-10-4140-kafka-0                                Running     2            2h36m     10.104.29.202     4am-node35     
fouram-op-10-4140-kafka-1                                Running     2            2h36m     10.104.15.64      4am-node20     
fouram-op-10-4140-kafka-2                                Running     1            2h36m     10.104.31.221     4am-node34     
fouram-op-10-4140-kafka-zookeeper-0                      Running     0            2h36m     10.104.31.214     4am-node34     
fouram-op-10-4140-kafka-zookeeper-1                      Running     0            2h36m     10.104.29.203     4am-node35     
fouram-op-10-4140-kafka-zookeeper-2                      Running     0            2h36m     10.104.15.65      4am-node20     
fouram-op-10-4140-minio-64d48475cf-w9xtb                 Running     0            2h36m     10.104.31.216     4am-node34     
 (common_func.py:407)
[2024-03-26 10:46:16,193 -  INFO - fouram]: [endpoint] Get the endpoint:fouram-op-10-4140-milvus.qa-milvus:19530 of fouram-op-10-4140 (operator_client.py:147)
[2024-03-26 10:46:16,194 -  INFO - fouram]: [Base] Service upgraded successfully:fouram-op-10-4140 (base.py:247)
[2024-03-26 10:46:16,197 -  INFO - fouram]: [check_params] scene_concurrent_locust input params: {'collection_params': {'shards_num': 2}, 'dataset_params': {'dataset_name': 'sift', 'dim': 128, 'dataset_size': '20m', 'ni_per': 50000, 'metric_type': 'L2'}, 'index_params': {'index_type': 'HNSW', 'index_param': {'M': 8, 'efConstruction': 200}}, 'load_params': {}, 'release_params': {'release_of_reload': True}, 'concurrent_params': {'concurrent_number': 50, 'during_time': '2h', 'interval': 60}, 'concurrent_tasks': [{'type': 'search', 'weight': 30, 'params': {'nq': 100, 'top_k': 100, 'search_param': {'ef': 128}, 'timeout': 120}}, {'type': 'flush', 'weight': 1, 'params': {'timeout': 120}}, {'type': 'upsert', 'weight': 19, 'params': {'nb': 200, 'start_id': 0, 'random_id': True, 'random_vector': True, 'timeout': 120}}]} (params_check.py:15)
[2024-03-26 10:46:16,198 -  INFO - fouram]: [check_params] scene_concurrent_locust required params: {'dataset_params': {'metric_type': 'L2', 'dim': 128, 'dataset_name': 'sift', 'dataset_size': '20m', 'ni_per': 50000}, 'collection_params': {'shards_num': 2}, 'load_params': {}, 'release_params': {'release_of_reload': True}, 'index_params': {'index_type': 'HNSW', 'index_param': {'M': 8, 'efConstruction': 200}}, 'concurrent_params': {'concurrent_number': 50, 'during_time': '2h', 'interval': 60}, 'concurrent_tasks': [{'type': 'search', 'weight': 30, 'params': {'nq': 100, 'top_k': 100, 'search_param': {'ef': 128}, 'timeout': 120}}, {'type': 'flush', 'weight': 1, 'params': {'timeout': 120}}, {'type': 'upsert', 'weight': 19, 'params': {'nb': 200, 'start_id': 0, 'random_id': True, 'random_vector': True, 'timeout': 120}}]} (params_check.py:33)
[2024-03-26 10:46:16,199 -  INFO - fouram]: [ConcurrentClientBase] The detailed test steps are as follows: 
        1. create a collection or use an existing collection
        2. build indexes on vector and scalar columns
        3. insert a certain number of vectors
        4. flush collection
        5. build indexes on vector and scalar columns with the same parameters
        6. count the total number of rows
        7. load collection
        8. call the go program to perform search concurrent operations
        9. clean all collections or not
         (concurrent_cases.py:460)
[2024-03-26 10:46:16,217 -  INFO - fouram]: [Base] Connection params: {'alias': 'default', 'host': 'fouram-op-10-4140-milvus.qa-milvus', 'port': '19530', 'uri': '', 'secure': False, 'user': '', 'password': '', 'token': '', 'db_name': ''} (base.py:237)
[2024-03-26 10:46:16,265 -  INFO - fouram]: [Base] Connect collection fouram_SUy5ONbj (base.py:277)
[2024-03-26 10:46:16,270 -  INFO - fouram]: [Base] Collection schema: {'auto_id': False, 'description': '', 'fields': [{'name': 'id', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float_vector', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}], 'enable_dynamic_field': False} (base.py:326)
[2024-03-26 10:46:16,270 -  INFO - fouram]: [CommonCases] Prepare collection fouram_SUy5ONbj done. (common_cases.py:76)
[2024-03-26 10:46:16,270 -  INFO - fouram]: [Base] Start release collection fouram_SUy5ONbj (base.py:321)
[2024-03-26 10:47:09,092 - ERROR - fouram]: RPC error: [release_collection], <MilvusException: (code=65535, message=empty grpc client: find no available querycoord, check querycoord state)>, <Time:{'RPC start': '2024-03-26 10:46:16.270340', 'RPC error': '2024-03-26 10:47:09.092472'}> (decorators.py:139)
[2024-03-26 10:47:09,093 - ERROR - fouram]: (api_response) : [Collection.release] <MilvusException: (code=65535, message=empty grpc client: find no available querycoord, check querycoord state)>, [requestId: 125e8976-eb5e-11ee-b239-06e37d66ade6] (api_request.py:57)
[2024-03-26 10:47:09,093 - ERROR - fouram]: [CheckFunc] release request check failed, response:<MilvusException: (code=65535, message=empty grpc client: find no available querycoord, check querycoord state)> (func_check.py:54)
FAILED[2024-03-26 10:47:09,364 -  INFO - fouram]: ********************************************* teardown ********************************************* (base.py:73)
``` 

### Steps To Reproduce

```markdown
https://argo-workflows.zilliz.cc/archived-workflows/qa/a8026997-6de7-4947-b6f1-96132f580145?nodeId=fouramf-concurrent-upgrade-config-4b-3807750563
```

### Milvus Log

pods:
```
fouram-op-10-4140-etcd-0                                          1/1     Running                           0               4h46m   10.104.26.123   4am-node32   <none>           <none>
fouram-op-10-4140-kafka-0                                         1/1     Running                           2 (4h45m ago)   4h46m   10.104.29.202   4am-node35   <none>           <none>
fouram-op-10-4140-kafka-1                                         1/1     Running                           2 (4h45m ago)   4h46m   10.104.15.64    4am-node20   <none>           <none>
fouram-op-10-4140-kafka-2                                         1/1     Running                           1 (4h45m ago)   4h46m   10.104.31.221   4am-node34   <none>           <none>
fouram-op-10-4140-kafka-zookeeper-0                               1/1     Running                           0               4h46m   10.104.31.214   4am-node34   <none>           <none>
fouram-op-10-4140-kafka-zookeeper-1                               1/1     Running                           0               4h46m   10.104.29.203   4am-node35   <none>           <none>
fouram-op-10-4140-kafka-zookeeper-2                               1/1     Running                           0               4h46m   10.104.15.65    4am-node20   <none>           <none>
fouram-op-10-4140-milvus-standalone-54897698d6-cjcsd              1/1     Running                           0               132m    10.104.16.15    4am-node21   <none>           <none>
fouram-op-10-4140-minio-64d48475cf-w9xtb                          1/1     Running                           0               4h46m   10.104.31.216   4am-node34   <none>           <none>
```

**Comment by xiaofan-luan** [comment]
I will try to refine the retry mechanism but we still need to invesigtate on why querycoord crash
/assign @congqixia

**Comment by congqixia** [comment]
milvus standalone did not crash during test period

**Comment by congqixia** [comment]
The root cause was that the first deployment deleted the session key of second deployment, which caused proxy could never find the querycoord turned active from standby state

**Comment by congqixia** [comment]
patch merged, could you please verify?
/unassign
/assign @ThreadDao

**Comment by ThreadDao** [comment]
fixed `master-20240401-b6fefee0c-amd64`
https://argo-workflows.zilliz.cc/archived-workflows/qa/8fd73359-2035-435f-a8fb-407b92823cac?nodeId=fouramf-concurrent-upgrade-config-10

**Comment by ThreadDao** [comment]
fixed `2.4-20240402-e69ae3bba-amd64`
